src/dc_receive_imf.rs

```diff
@@ -2197,8 +2198,8 @@ async fn dc_add_or_lookup_contacts_by_address_list(
     address_list: &[SingleInfo],
     origin: Origin,
     prevent_rename: bool,
-) -> Result<ContactIds> {
-    let mut contact_ids = ContactIds::new();
+) -> Result<Vec<u32>> {
+    let mut contact_ids = BTreeSet::new();
```

Comment:
Why not directly use a `Vec<u32>` here? I mean, appending elements to a `Vec` shouldn't be more expensive than to a `BTreeSet`, and then we can directly return it at the end.

Reply:
In this case we need to check if the vector does not contain the contact ID already before inserting it, because the function documentation guarantees that IDs are unique. Not a big deal even for 100 users, but still not nice to have an algorithm that is quadratic in time in the message reception pipeline.

Reply:
Makes sense